tests/subsys/bluetooth/mesh/scheduler_model/action_planning/src/main.c

```diff
@@ -0,0 +1,665 @@
+{
+	struct sched_evt *evt = sched_evt_next();
+	struct test_scene_srv *ptr = CONTAINER_OF(srv, struct test_scene_srv, srv);
+
+	zassert_equal(evt->elem_idx, ptr->elem_idx, "Wrong sequence of scene events");
+	zassert_equal(BT_MESH_SCHEDULER_SCENE_RECALL, evt->evt, "Incorrect scene event");
+	zassert_equal(evt->transition_time, transition->time, "Unexpected transition time");
+	zassert_equal(evt->scene_number, scene, "Incorrect Scene number");
+
+	if (fired_tm) {
+		fired_tm[galloc_cnt] = *bt_mesh_time_srv_localtime(&time_srv, k_uptime_get());
+		galloc_cnt++;
+	}
+
+	if (is_sched_evt_empty()) {
+		k_sem_give(&action_fired);
+	}
+	return 0;
+}
+
+struct tm *bt_mesh_time_srv_localtime(struct bt_mesh_time_srv *srv,
+				      int64_t uptime)
+{
+	static struct tm timeptr;
+	struct bt_mesh_time_tai tai;
+
+	zassert_ok(ts_to_tai(&tai, &start_tm), "cannot convert tai time");
+
+	int64_t tmp =  tai_to_ms(&tai);
+
+	tmp += uptime;
+	tai = tai_at(tmp);
+	tai_to_ts(&tai, &timeptr);
+
+	return &timeptr;
+}
+
+void bt_mesh_model_msg_init(struct net_buf_simple *msg, uint32_t opcode)
+{
+	net_buf_simple_init(msg, 0);
+}
+
+int64_t bt_mesh_time_srv_mktime(struct bt_mesh_time_srv *srv, struct tm *timeptr)
+{
+	return (timeutil_timegm64(timeptr) - timeutil_timegm64(&start_tm)) * 1000;
+}
+
+int bt_mesh_scene_srv_pub(struct bt_mesh_scene_srv *srv,
+			 struct bt_mesh_msg_ctx *ctx)
+{
+	return 0;
+}
+
+int bt_mesh_onoff_srv_pub(struct bt_mesh_onoff_srv *srv,
+			  struct bt_mesh_msg_ctx *ctx,
+			  const struct bt_mesh_onoff_status *status)
+{
+	return 0;
+}
+
+void bt_mesh_time_encode_time_params(struct net_buf_simple *buf,
+				     const struct bt_mesh_time_status *status)
+{
+}
+
+int bt_mesh_msg_send(struct bt_mesh_model *model, struct bt_mesh_msg_ctx *ctx,
+	       struct net_buf_simple *buf)
+{
+	return 0;
+}
+
+int _bt_mesh_time_srv_update_handler(struct bt_mesh_model *model)
+{
+	return 0;
+}
+/* Redefined mocks */
+
+static void setup(void)
+{
+	mod_elem_idx_prep();
+
+	/* 1st of Jan 2010 */
+	start_tm.tm_year = 110;
+	start_tm.tm_mon = 0;
+	start_tm.tm_mday = 1;
+	start_tm.tm_wday = 5;
+	start_tm.tm_hour = 0;
+	start_tm.tm_min = 0;
+	start_tm.tm_sec = 0;
+
+	gfire_cnt = 0;
+	galloc_cnt = 0;
+	fired_tm = NULL;
+
+	k_sem_reset(&action_fired);
+	zassert_not_null(_bt_mesh_scheduler_srv_cb.init,
+			 "Init cb is null");
+
+	_bt_mesh_scheduler_srv_cb.init(sched_mod_elem1);
+	_bt_mesh_scheduler_srv_cb.init(sched_mod_elem4);
+}
+
+static void teardown(void)
+{
+	if (fired_tm) {
+		free(fired_tm);
+	}
+	zassert_not_null(_bt_mesh_scheduler_srv_cb.reset,
+			 "Reset cb is null");
+
+	_bt_mesh_scheduler_srv_cb.reset(sched_mod_elem1);
+	_bt_mesh_scheduler_srv_cb.reset(sched_mod_elem4);
+}
+
+static void action_put(const struct bt_mesh_schedule_entry *test_action,
+		       struct bt_mesh_model *sched_mod)
+{
+	BT_MESH_MODEL_BUF_DEFINE(buf, BT_MESH_SCHEDULER_OP_ACTION_SET_UNACK,
+				 BT_MESH_SCHEDULER_MSG_LEN_ACTION_SET);
+
+	net_buf_simple_init(&buf, 0);
```

Comment:
Small comment: I believe `net_buf_simple_init` is not necessary when using `BT_MESH_MODEL_BUF_DEFINE`, only with `NET_BUF_SIMPLE` :) 
```suggestion
```